test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJstack.java

```diff
@@ -0,0 +1,236 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.test.attachAPI;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.lang.Thread.State;
+import java.lang.management.ThreadInfo;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.openj9.test.util.StringPrintStream;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import openj9.tools.attach.diagnostics.info.JvmThreadInfo;
+import openj9.tools.attach.diagnostics.info.ThreadGroupInfo;
+
+@Test(groups = { "level.extended" })
+public class TestJstack extends AttachApiTest {
+
+	private static final String JSTACK_COMMAND = "jstack"; //$NON-NLS-1$
+	Object syncObject = new Object();
+
+	@Test
+	public void testSingleThread() throws IOException {
+		try {
+			Thread testThread = new Thread(new Runnable() {
```

Comment:
This could use a method handle: `this::method1`.

Reply:
Cool.  Thanks.